app/models/thread.php

```diff
@@ -23,6 +23,10 @@ public static function getAll($offset, $limit)
         foreach ($rows as $row) {
             $threads[] = new self($row);
         }
+
+        if (!$rows) {
+            throw new RecordNotFoundException('No Record found');
```

Comment:
I think this if block must be placed before the foreach? please check the value first before create array of Thread objects.. :bow: